app/models/nic/bmc.rb

```diff
@@ -39,6 +37,24 @@ def ensure_physical
     def enc_attributes
       @enc_attributes ||= (super + %w(username password provider))
     end
+
+    private
+
+    def bmc_proxy
+      if subnet.present?
+        proxy = subnet.proxies.find { |subnet_proxy| subnet_proxy.has_feature?('BMC') }
```

Comment:
It was the underlying associations I was a bit worried about, but in my tests it seems that the default taxonomy scope of smart proxy doesn't seem to apply to `subnet.dns` etc., so I guess that's OK.